code/modules/mob/living/simple_animal/bot_swarm/swarmer.dm

```diff
@@ -0,0 +1,598 @@
+////Deactivated swarmer shell////
+/obj/item/unactivated_swarmer
+	name = "unactivated swarmer"
+	desc = "A currently unactivated swarmer. Swarmers can self activate at any time, it would be wise to immediately dispose of this."
+	icon = 'icons/mob/swarmer.dmi'
+	icon_state = "swarmer_unactivated"
+
+/obj/item/unactivated_swarmer/New()
+	notify_ghosts("An unactivated swarmer has been created in [get_area(src)]! <a href=?src=\ref[src];ghostjoin=1>(Click to enter)</a>")
+	..()
+
+/obj/item/unactivated_swarmer/Topic(href, href_list)
+	if(href_list["ghostjoin"])
+		var/mob/dead/observer/ghost = usr
+		if(istype(ghost))
+			attack_ghost(ghost)
+
+/obj/item/unactivated_swarmer/attack_ghost(mob/user)
+	var/be_swarmer = alert("Become a swarmer? (Warning, You can no longer be cloned!)",,"Yes","No")
+	if(be_swarmer == "No")
+		return
+	if(qdeleted(src))
+		user << "Swarmer has been occupied by someone else."
+		return
+	var/mob/living/simple_animal/hostile/swarmer/S = new /mob/living/simple_animal/hostile/swarmer(get_turf(loc))
+	S.key = user.key
```

Comment:
@tigercat2000 acutally was looking at the mind datum file for transfer_to
>When creating a new mob for an existing IC character (e.g. cloning a dead guy or borging a brain of a human)
>		the existing mind of the old mob should be transfered to the new mob like so:
>			mind.transfer_to(new_mob)
>	-	You must not assign key= or ckey= after transfer_to() since the transfer_to transfers the client for you.
>		By setting key or ckey explicitly after transfering the mind with transfer_to you will >cause bugs like DCing
>		the player.
>	-	IMPORTANT NOTE 2, if you want a player to become a ghost, use mob.ghostize() It does all the hard work for you.
>	-	When creating a new mob which will be a new IC character (e.g. putting a shade in a construct or randomly selecting
>		a ghost to become a xeno during an event). Simply assign the key or ckey like you've always done.
>			new_mob.key = key
>		The Login proc will handle making a new mob for that mobtype (including setting up >stuff like mind.name). Simple!
>		However if you want that mind to have any special properties like being a traitor etc >you will have to do that
>		yourself.